tests/Service/DatabaseManagerTest.php

```diff
@@ -140,6 +161,33 @@ public function testIsCreatingAndRollingBackTransaction(): void
         $this->assertTrue($db->rollBack());
     }
 
+    public function testIsCreatingAndRollingBackTransactionUsingExecuteTransaction(): void
+    {
+        $this->expectException(DatabaseException::class);
+
+        $db = new DatabaseManager();
+        $connectionOptions = [
+            'host' => TEST_DATABASE_HOST,
+            'port' => TEST_DATABASE_PORT,
+            'dbname' => TEST_DATABASE_DBNAME,
+            'username' => TEST_DATABASE_USERNAME,
+            'password' => TEST_DATABASE_PASSWORD,
+        ];
+        $db->addConnection(DatabaseManager::DRIVER_MYSQL, $connectionOptions, DatabaseManager::ROLE_MASTER);
+        $db->addConnection(DatabaseManager::DRIVER_MYSQL, $connectionOptions, DatabaseManager::ROLE_SLAVE);
+
+        $db->execute('CREATE TABLE testing_rollback (id int(11))');
```

Comment:
This test is going to fail after the first time because the table would already exist wouldn't it?

Reply:
It makes sense, but I don't know why it's working even after the first time. The `PdoAdapterTest` is also doing the same thing. Somehow the test is undoing what it does in the teardown.

Reply:
You are right! When I moved the `$this->expectException(DatabaseException::class);` I got the error of the table that already exists! hahahaha